### PR TITLE
[2.12] excluding delete system index tests from a security enabled cluster

### PR DESCRIPTION
**Pull request by joshpalis**
### Description
Excludes integration test classes that attempt to delete system indices from running when security is enabled
 
### Issues Resolved
Part of https://github.com/opensearch-project/skills/issues/189
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dbwiddis — APPROVED** [review]
This looks familiar.

**Review by owaiskazi19 — APPROVED** [review]
LGTM! Thanks @joshpalis

File: build.gradle [hunk 1]
```diff
@@ -429,12 +358,24 @@ integTest {
         is_https = is_https == null ? "true" : is_https
         user = user == null ? "admin" : user
         password = password == null ? "admin" : password
```
Comment (dhrubo-os) [inline build.gradle:360]: According to [this issue](https://github.com/opensearch-project/security/issues/3624), don't we need to change the default password from `admin`?

File: build.gradle [hunk 2]
```diff
@@ -463,7 +404,91 @@ testClusters.integTest {
...
+                        }
+                    }
+                }))
+            }
+        }
+
+        getNodes().forEach { node ->
+            var creds = node.getCredentials()
+            if (creds.isEmpty()) {
+                creds.add(Map.of('username', 'admin', 'password', 'admin'))
```
Comment (dhrubo-os) [inline build.gradle:426]: same here

File: build.gradle (hunk 1 above)
Comment (joshpalis) [reply build.gradle:360]: no since we're not directly using the install demo configuration script

Comment (joshpalis) [reply build.gradle:360]: Reference : https://github.com/opensearch-project/flow-framework/issues/310#issuecomment-1908844576

Comment (dhrubo-os) [reply build.gradle:360]: I see, thanks for the reference. Let's add the comment in the code as this is bit confusing.

**Comment by owaiskazi19** [comment]
@dhrubo-os any concern before we merge this?

**Comment by dhrubo-os** [comment]
> @dhrubo-os any concern before we merge this?

no major concern. Let's just add the comment as it was also suggested by others.

**Review by dhrubo-os — APPROVED** [review]

**Pull request merged**